julia_con/RelatedCon/src/RelatedCon.jl

```diff
@@ -1,33 +1,29 @@
 module RelatedCon
 
-using JSON3
-using StructTypes
-using Dates
-using PrecompileTools
-using ChunkSplitters
+using JSON3, StructTypes, Dates, StaticArrays, ChunkSplitters
 using Base.Threads: @threads, nthreads
 
-const topn = 5
-
 export main
 
+const topn = 5
+
 struct PostData
     _id::String
     title::String
-    tags::Vector{Symbol}
+    tags::Vector{String}
 end
 
 struct RelatedPost
     _id::String
-    tags::Vector{Symbol}
-    related::NTuple{topn, PostData}
+    tags::Vector{String}
+    related::SVector{topn, PostData}
 end
 
 StructTypes.StructType(::Type{PostData}) = StructTypes.Struct()
 
-function maxindex!(xs::Vector, maxs)
+function fastmaxindex!(xs::Vector{T}, topn, maxs) where {T}
     # each element is a pair idx => val
-    maxs .= (1 => 0)
+    maxs .= (one(T) => zero(T))
```

Comment:
is there any reason we use this `::Pair` implementation instead of two `MVector` like the non-concurrent version?

Reply:
I'm not sure actually, I think @MasonProtter changed it, it seems to me they require less indexing operations in fastmaxindex, I assumed that if it was changed it should be better. On my machine they look identical performance wise

Reply:
Either way, the first element should always be an `Int`, not a `one(T)` since it's an index to an array.

Reply:
is it okay to return to previous methodology @MasonProtter ? I think it doesn't change much but for consistency we could use one or the other in both versions

Reply:
if you have the time you can maybe look into it to see if it does change something on your macines

Reply:
upon further testing, I see that it is maybe a bit faster the other way so I restored that, hopefully we all agree that it is okay to use it